test/testenv/verificationutils.go

```diff
@@ -1049,3 +1049,17 @@ func VerifyAppRepoState(ctx context.Context, deployment *Deployment, testenvInst
 		return int(appDeploymentInfo.RepoState)
 	}, deployment.GetTimeout(), PollInterval).Should(gomega.Equal(repoValue))
 }
+
+// VerifyIsDeploymentInProgressFlagIsSet verify IsDeploymentInProgress flag is set to true
```

Comment:
This will be tricky to test what happens if by the time we check status has moved to next stage

Reply:
I ran the test quite a few times with no failure, will keep an eye out for future failures.